Tasks/NuGetCommandV2/nugetcustom.ts

```diff
@@ -35,12 +37,12 @@ export async function run(nuGetPath: string): Promise<void> {
     }
 
     try {
-        let credProviderPath = nutil.locateCredentialProvider();
-
         // Clauses ordered in this way to avoid short-circuit evaluation, so the debug info printed by the functions
         // is unconditionally displayed
         const quirks = await ngToolRunner.getNuGetQuirksAsync(nuGetPath);
-        const useCredProvider = ngToolRunner.isCredentialProviderEnabled(quirks) && credProviderPath;
+        const credProviderPath: string = nutil.locateCredentialProvider(quirks);
+        const useV1CredProvider: string = ngToolRunner.isCredentialProviderV1Enabled(quirks) ? credProviderPath : null;
```

Comment:
Checking that credProviderPath exists should still be there. The name of these variables suggest they should be booleans, I would keep it that way and assign the string below when creating the environmentSettings.

Reply:
I was confused about that as well. The original way using `ngToolRunner.isCredentialProviderEnabled(quirks) && credProviderPath;` doesn't return a boolean. It returns the value of credProviderPath as a string. If I turned it into following it would be a boolean: `ngToolRunner.isCredentialProviderEnabled(quirks) && !!credProviderPath`. 

The reason I turned it into following is to make it more clear that it's a string, not a boolean: `ngToolRunner.isCredentialProviderV1Enabled(quirks) ? credProviderPath : null;`

`useCredProvider `in `authInfo `below is also a string instead of a boolean. 

I could either rename `useV1CredProvider `and `useV2CredProvider `so that they seem more like a string, or, I could refactor `NuGetExtendedAuthInfo `to expect a boolean for `useCredProvider `instead of a string. Any preferences?